docs/summer-of-code-ideas/summer-of-code-reports/gsoc-2019-abelbriggs.md

```diff
@@ -0,0 +1,168 @@
+ reverse-engineered driver [nouveau](https://nouveau.freedesktop.org/wiki/).
+ 
+### Enhance GraphicsFuzz's shader generator - GLSL built-in support
+
+When GraphicsFuzz generates fuzzed/garbage expressions with the assumption that they won't be
+executed (e.g. in `(true ? x : y)`, y will never be executed), it is able to generate calls to GLSL
+built-in functions(think `abs()`, `sqrt()`, etc.) with garbage values as arguments. To do this, however,
+GraphicsFuzz needs to know what types to fuzz expressions for, or it will cause syntax errors.
+The following PRs involve cross-checking built-in functions with the language version they were
+introduced in, then adding their function prototypes to GraphicsFuzz.
+
+[#521](https://github.com/google/graphicsfuzz/pull/521):
+Add built-in support for GLSL Integer Functions
+
+[#528](https://github.com/google/graphicsfuzz/pull/528):
+Add built-in function support for GLSL Vector Relational Functions
+
+[#549](https://github.com/google/graphicsfuzz/pull/549):
+Add built-in function support for GLSL 3.20 Fragment Processing Functions
+
+[#563](https://github.com/google/graphicsfuzz/pull/563):
+Refactor exponential builtins into separate function
+
+Additionally, minor issues arised from adding support for these functions - some of these functions
+involve using `out` parameters and modify said parameters during their execution. This would
+potentially cause 'side effects' - lvalues being modified when not expected. The following PR
+relates to mitigating this issue.
+
+[#533](https://github.com/google/graphicsfuzz/pull/533):
+Check for side effects with lvalue parameters of built-in functions
+
+#### Leftovers
+
+The following issues are nits or minor problems related to built-ins that were left unfixed due to 
+time or knowledge constraints.
+
+[#550](https://github.com/google/graphicsfuzz/issues/550):
+Ensure certain function prototypes can use non-uniform shader input variables
+
+[#570](https://github.com/google/graphicsfuzz/issues/570):
+Be less conservative about when FunctionCallExprTemplates yield expressions that have side effects.
+
+### Enhance GraphicsFuzz's shader generator - identities and opaque value generation
+
+Among the various operations that GraphicsFuzz's generator can do to a shader when fuzzing, two of
+the most important are identity transformations and opaque value generation.
+
+An identity transformation is a function that, given an expression `e`, transforms `e -> f`, where `f` is
+an expression equivalent for all intents and purposes to `e` (*semantically equivalent*). GraphicsFuzz
+is able to use various properties of GLSL data types and built-in functions to craft these identity
+transformations, and can recursively apply them to create obscenely complicated expressions.
+
+For a toy example of an identity: Let `e` be a float of any valid value and let `identity(x) = x * 1.0`. 
+Then `e` is semantically equivalent to `identity(e)`.
+
```

Comment:
Before "is not obvious to a compiler" I would add something to indicate that the rvalue is guaranteed to have a certain value, but that that value is not obvious to a compiler.

Reply:
Fixed.